src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -11,62 +11,37 @@
 
 namespace Symfony\Component\DependencyInjection;
 
+use Psr\Container\ContainerExceptionInterface;
 use Psr\Container\ContainerInterface as PsrContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
 use Symfony\Component\DependencyInjection\Exception\RuntimeException;
 use Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException;
 use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Contracts\Service\ServiceLocatorTrait;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
 
 /**
  * @author Robin Chalas <robin.chalas@gmail.com>
  * @author Nicolas Grekas <p@tchwork.com>
  */
 class ServiceLocator implements PsrContainerInterface
 {
-    private $factories;
-    private $loading = array();
-    private $externalId;
-    private $container;
-
-    /**
-     * @param callable[] $factories
-     */
-    public function __construct(array $factories)
-    {
-        $this->factories = $factories;
+    use ServiceLocatorTrait {
+        get as doGet;
```

Comment:
does this make `doGet` public or no ?

Reply:
nope, visibility changes must be explicit.

Reply:
ok, got it, fixed :)